### PR TITLE
Fix flag conflict

### PR DESCRIPTION
**Pull request by fahedouch**
we cannot redefine rootCmd Persistent alias in subcommands. So the rootCmd alias that conflicts with subcommands alias should be added to `rootCmd.Flags()` and not `rootCmd.PersistentFlags()`  ( `rootCmd.Flags()`  is full set of flags  which include `rootCmd.PersistentFlags()`)

fixing https://github.com/containerd/nerdctl/issues/1334

Signed-off-by: fahed dorgaa <fahed.dorgaa@gmail.com>

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Can we have a test?

**Comment by fahedouch** [comment]
@AkihiroSuda PTAL

File: cmd/nerdctl/main.go [hunk 1]
```diff
@@ -161,34 +161,36 @@ func NewConfig() *Config {
...
+	aliasToBeInherited := pflag.NewFlagSet(rootCmd.Name(), pflag.ExitOnError)
+
 	rootCmd.PersistentFlags().Bool("debug", cfg.Debug, "debug mode")
 	rootCmd.PersistentFlags().Bool("debug-full", cfg.DebugFull, "debug mode (with full output)")
-	// -a is nonPersistentAlias (conflicts with nerdctl images -a)
-	AddPersistentStringFlag(rootCmd, "address", []string{"host"}, []string{"a", "H"}, cfg.Address, "CONTAINERD_ADDRESS", `containerd address, optionally with "unix://" prefix`)
-	// -n is nonPersistentAlias (conflicts with nerdctl logs -n)
-	AddPersistentStringFlag(rootCmd, "namespace", nil, []string{"n"}, cfg.Namespace, "CONTAINERD_NAMESPACE", `containerd namespace, such as "moby" for Docker, "k8s.io" for Kubernetes`)
+	// -a is aliases (conflicts with nerdctl images -a)
+	AddPersistentStringFlag(rootCmd, "address", []string{"a, H"}, nil, []string{"host"}, aliasToBeInherited, cfg.Address, "CONTAINERD_ADDRESS", `containerd address, optionally with "unix://" prefix`)
```
Comment (junnplus) [inline cmd/nerdctl/main.go:185]: why is `"a, H"`, instead of `"a", "H"`?

Comment (fahedouch) [reply cmd/nerdctl/main.go]: fixed

File: cmd/nerdctl/main.go [hunk 2]
```diff
@@ -432,7 +441,7 @@ func AddStringFlag(cmd *cobra.Command, name string, aliases []string, value stri
 
 // AddPersistentStringFlag is similar to AddStringFlag but persistent.
 // See https://github.com/spf13/cobra/blob/master/user_guide.md#persistent-flags to learn what is "persistent".
-func AddPersistentStringFlag(cmd *cobra.Command, name string, aliases, nonPersistentAliases []string, value string, env, usage string) {
```
Comment (junnplus) [inline cmd/nerdctl/main.go:435]: It seems that we are not using `localAliases`

Comment (fahedouch) [reply cmd/nerdctl/main.go:435]: we should keep it to differentiate the 3 types of aliases to avoid confusion.

**Review by junnplus — APPROVED** [review]
Thanks, LGTM

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**